### ISSUE TITLE
Missing parenthesis for typedef function pointer

### ISSUE THREAD

**Issue by leandrohrb** [doxygen/doxygen#7514]
Hello :)

After **upgrading Doxygen from 1.8.16 to 1.8.17**, the building process of Weston's documentation
stopped working because of the function bellow:

```c
// file weston-log.h
// https://gitlab.freedesktop.org/wayland/weston/blob/master/include/libweston/weston-log.h

/** weston_log_scope callback
 *
 * @param sub The subscription.
 * @param user_data The \c user_data argument given to
 * weston_compositor_add_log_scope()
 *
 * @memberof weston_log_scope
 */
typedef void (*weston_log_scope_cb)(struct weston_log_subscription *sub,
				    void *user_data);
```

Looking at the xml files generated by Doxygen 1.8.17, **a missing parenthesis is
noticeable** in structweston__log__scope.xml. Adding the parenthesis
makes everything work as usual.

#### in 1.8.17
```
<definition>typedef void(* weston_log_scope_cb(struct weston_log_subscription *sub, void *user_data)</definition>
```

##### in 1.8.16
```
<definition>typedef void(* weston_log_scope_cb) (struct weston_log_subscription *sub, void *user_data)</definition>
```

Here it is the zip to reproduce the error, containing a brand-new config file
(only edited PROJECT_NAME, OUTPUT, INPUT and GENERATE_XML) and the source
files.

[reproduce-error.zip](https://github.com/doxygen/doxygen/files/4068212/reproduce-error.zip)

If there's anything else that I can share to help, let me know.
Thank you!

**Comment by albert-github** [comment]
A test on the given case gives (just use: `grep typedef output/xml/structweston__log__scope.xml | grep definition` on the generated output) that the version:

> Commit: 4e21d19 [4e21d19]
> Parents: 40f187c, 74c7407
> Date: Saturday, September 28, 2019 12:08:11 PM
> Merge pull request #7282 from albert-github/feature/issue_7274

is still OK, but the next commit in master:

> Commit: d2b8ae1 [d2b8ae1]
> Parents: 4e21d19, ae0a5ec
> Date: Monday, October 7, 2019 9:02:04 PM
> Merge branch 'std'

shows the different behavior, so looks like a regression when introducing "Use smartpointers to manage the lifetime of Entry objects".
Note also the current master shows still the problem.

In other words the problem has the same commit origin as #7508

**Comment by doxygen** [comment]
I've push [this commit](https://github.com/doxygen/doxygen/commit/cbeaf5763aef97b1f116c9996cb69c74133162bc) to fix this problem. Please let me know if it works for you.

**Comment by leandrohrb** [comment]
I confirm that it works for me. 

Thanks a lot for the fix :)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.18.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
